src/UI/TreeView.re

```diff
@@ -81,23 +89,69 @@ module Make = (Model: TreeModel) => {
       style=Style.[width(Constants.arrowSize), height(Constants.arrowSize)]
     />;
 
+  // COunts the number of axpanded nodes before the node specified by the given path
+  let nodeOffsetByPath = (tree, path) => {
+    let rec loop = (node, path) =>
+      switch (path) {
+      | [] => failwith("Well, this is awkward (ie. unreachable)")
+      | [focus, ...focusTail] =>
+        if (focus != node) {
+          `NotFound(Model.expandedSubtreeSize(node));
+        } else {
+          switch (Model.kind(node)) {
+          | `Node(`Closed)
+          | `Leaf => `Found(0)
+
+          | `Node(`Open) =>
+            let rec loopChildren = (count, children) =>
+              switch (children) {
+              | [] => `NotFound(count)
+              | [child, ...childTail] =>
+                switch (loop(child, focusTail)) {
+                | `Found(subtreeCount) => `Found(count + subtreeCount)
+                | `NotFound(subtreeCount) =>
+                  loopChildren(count + subtreeCount, childTail)
+                }
+              };
+            loopChildren(1, Model.children(node));
+          };
+        }
+      };
+
+    switch (loop(tree, path)) {
+    | `Found(count) => Some(count)
+    | `NotFound(_) => None
+    };
+  };
```

Comment:
note to self: Try to factor out a more general tree traversal function from this and `findNodesByLocalPath`